Sources/App/Migrations/052/UpdateSearchAddProductType.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020-2021 Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Fluent
+import SQLKit
+
+
+struct UpdateSearchAddProductType: Migration {
+    let dropSQL: SQLQueryString = "DROP MATERIALIZED VIEW search"
+
+    func prepare(on database: Database) -> EventLoopFuture<Void> {
+        guard let db = database as? SQLDatabase else {
+            fatalError("Database must be an SQLDatabase ('as? SQLDatabase' must succeed)")
+        }
+
+        return db.raw(dropSQL).run()
+            .flatMap { db.raw("""
+            -- v6
+            CREATE MATERIALIZED VIEW search AS
+            SELECT
+              p.id AS package_id,
+              p.platform_compatibility,
+              p.score,
+              r.keywords,
+              r.last_commit_date,
+              r.license,
+              r.name AS repo_name,
+              r.owner AS repo_owner,
+              r.stars,
+              r.last_activity_at,
+              r.summary,
+              v.package_name,
+              pr.type
+            FROM packages p
+              JOIN repositories r ON r.package_id = p.id
+              JOIN versions v ON v.package_id = p.id
+              LEFT JOIN products pr ON pr.version_id = v.id
```

Comment:
One thing we need to consider here is that this will add duplicates to the `search` view.

Prior to this change, the search view has exactly as many rows as `packages`. For instance, with the recent prod dump `spi_prod-2022-06-14.dump` we've got:

```
select count(*) from packages;  -- 4672

-- v5
select count(*) from search;    -- 4672
-- v6
select count(*) from search;    -- 6835
```

The duplicates as packages that have multiple products. This is probably ok but we should double check that we're running a `DISTINCT` query for the final results.

Reply:
Not quite 😅

<img width="444" alt="CleanShot 2022-06-17 at 08 21 10@2x" src="https://user-images.githubusercontent.com/65520/174237639-34524572-5cbf-4213-b0fb-9038d5d44167.png">

Reply:
Will address it once I get search working again